### PR TITLE
refactor: Use `eslint-config-eslint` for linting this repo

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Updates eslint config to use `eslint-config-eslint`.

#### What changes did you make? (Give an overview)

* Updated `eslint.config.js` for this repo  to use `eslint-config-eslint`.
* Added `eslint-config-prettier` to disable conflicting rules.
* Fixed lint errors. Some changes are in the production code, so this is marked as "refactor".

#### Related Issues

No.

#### Is there anything you'd like reviewers to focus on?

I left two things for follow-ups (I'll open separate issues when this is merged):

1. `eslint-plugin-jsdoc` rules are disabled for now. Enabling these rules will need special attention because we're using JSDoc to generate types.
2. I'm not sure if we're getting all checks from `eslint-plugin-n` because linted files are not published.

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -1,12 +1,63 @@
...
+	eslintConfigPrettier,
+
+	{
+		rules: {
+			// disable additional deprecated rules
+			"padding-line-between-statements": "off",
+			"spaced-comment": "off",
+
+			// disable rules we don't want to use from eslint-config-eslint
+			"no-undefined": "off",
```
Comment (mdjermanovic) [inline eslint.config.js:25]: There were 34 problems reported by this rule. I assumed we don't want to use it in this repo.

**Comment by nzakas** [comment]
Rather than using `eslint-config-prettier`, could we just create a config in `eslint-config-eslint` that doesn't include formatting rules?

**Comment by mdjermanovic** [comment]
> Rather than using `eslint-config-prettier`, could we just create a config in `eslint-config-eslint` that doesn't include formatting rules?

Good idea! I'll make a PR for this in eslint/eslint.

**Comment by mdjermanovic** [comment]
Here's the PR: https://github.com/eslint/eslint/pull/18560

**Comment by mdjermanovic** [comment]
Updated to eslint-config-eslint v11 and removed eslint-config-prettier now.

**Review by nzakas — APPROVED** [review]
LGTM.

**Pull request merged**